src/components/MobileWalletDialog/routes/CreateWallet/ManualBackup.tsx

```diff
@@ -0,0 +1,144 @@
+import { FiCopy } from 'react-icons/fi'
+import { IoShieldCheckmark } from 'react-icons/io5'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router'
+import { CarouselDots } from 'components/CarouselDots/CarouselDots'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogCloseButton } from 'components/Modal/components/DialogCloseButton'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderLeft,
+  DialogHeaderMiddle,
+  DialogHeaderRight,
+} from 'components/Modal/components/DialogHeader'
+import { SlideTransition } from 'components/SlideTransition'
+import type { MobileLocationState } from 'context/WalletProvider/MobileWallet/types'
+
+import { MobileWalletDialogRoutes } from '../../types'
+
+const copyIcon = <Icon as={FiCopy} />
+
+type ManualBackupProps = {
+  showContinueButton?: boolean
+}
+
+export const ManualBackup = ({ showContinueButton = true }: ManualBackupProps) => {
+  const history = useHistory()
+  const location = useLocation<MobileLocationState>()
+  const translate = useTranslate()
+  const bgColor = useColorModeValue('blue.500', 'blue.500')
+
+  const handleBack = useCallback(() => {
+    if (showContinueButton) {
+      history.push(MobileWalletDialogRoutes.KeepSafe, { vault: location.state?.vault })
+      return
+    }
+
+    history.push(MobileWalletDialogRoutes.Saved)
+  }, [history, location.state?.vault, showContinueButton])
+
+  const handleContinue = useCallback(() => {
+    history.push(MobileWalletDialogRoutes.CreateBackupConfirm, { vault: location.state?.vault })
+  }, [history, location.state?.vault])
+
+  const words = useMemo(() => {
+    if (!location.state?.vault) return []
+
+    return location.state.vault.getWords() ?? []
+  }, [location.state?.vault])
+
+  const { onCopy, hasCopied } = useClipboard(words.join(' '))
+
+  return (
+    <SlideTransition>
+      <DialogHeader>
+        <DialogHeaderLeft>
+          <DialogBackButton onClick={handleBack} />
+        </DialogHeaderLeft>
+        <DialogHeaderMiddle>
+          <Box minWidth='40px'>
+            <CarouselDots length={3} activeIndex={2} />
+          </Box>
+        </DialogHeaderMiddle>
+        <DialogHeaderRight>
+          <DialogCloseButton />
+        </DialogHeaderRight>
+      </DialogHeader>
+      <DialogBody>
+        <VStack spacing={6} mb={6} alignItems='flex-start'>
+          <Box>
+            <CText fontSize='2xl' fontWeight='bold'>
+              {translate('walletProvider.manualBackup.header')}
+            </CText>
+            <CText color='text.subtle'>{translate('walletProvider.manualBackup.subHeader')}</CText>
+          </Box>
+
+          <Box bg={bgColor} borderRadius='xl' p={4} width='full'>
+            <SimpleGrid columns={2} spacing={2}>
+              {words.map((word, index) => (
```

Comment:
nitpick: can be memoized

Reply:
Done in [522ad70](https://github.com/shapeshift/web/pull/8760/commits/522ad7093231b9978ced58eabda1c32b15f7a001)